lib/intercom_flutter.dart

```diff
@@ -2,26 +2,52 @@ library intercom_flutter;
 
 import 'dart:async';
 
+import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
 
 enum IntercomVisibility { gone, visible }
 
+typedef void MessageHandler(Map<String, dynamic> message);
+
 class Intercom {
   static const MethodChannel _channel =
       const MethodChannel('maido.io/intercom');
+  static MessageHandler _messageHandler;
 
-  static Future<dynamic> initialize(
-    String appId, {
-    String androidApiKey,
-    String iosApiKey,
-  }) {
+  /// This is useful since end application don't need to store the token by itself.
+  /// It will be send through message handler so application can use it in any way it wants.
+  static String _iosDeviceToken;
+
+  static Future<dynamic> initialize(String appId,
+      {String androidApiKey, String iosApiKey, MessageHandler onMessage}) {
+    // Backward compatibility, show new feature in debug mode.
+    if (onMessage == null && !kReleaseMode) {
+      _messageHandler = (data) => print("[INTERCOM_FLUTTER] On message: $data");
+    }
+    _messageHandler = onMessage;
```

Comment:
I think this is missing an `else` clause, you're overriding `_messageHandler` with null if it's not defined

Reply:
You are right, missing else if clause.